uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js

```diff
@@ -333,28 +307,61 @@ define([
          *  - timestamps (Date object is used by some 3rd party directives)
          *  - date format we get from server
          */
-        var formatted = moment(date, [HR_settings.DATE_FORMAT.toUpperCase(), 'x', 'YYYY-MM-DD']);
+        var formatted = moment(date, [HRSettings.DATE_FORMAT.toUpperCase(), 'x', 'YYYY-MM-DD']);
 
         return (formatted.isValid()) ? formatted.format('YYYY-MM-DD') : null;
       };
 
-      $scope.dpOpen = function ($event, name) {
+      vm.dpOpen = function ($event, name) {
         $event.preventDefault();
         $event.stopPropagation();
-        $scope.dpOpened[name] = true;
+        vm.dpOpened[name] = true;
       };
 
-      $scope.dropzoneClick = function () {
+      vm.dropzoneClick = function () {
         $timeout(function () {
           document.getElementById($rootScope.prefix + 'document-files').click();
         });
       };
 
       // Display help message
-      $scope.remindMeInfo = function () {
-        CRM.help('Remind me?', $scope.remindMeMessage, 'error');
+      vm.remindMeInfo = function () {
+        CRM.help('Remind me?', vm.remindMeMessage, 'error');
+      };
+
+      /**
+       * Open the given file and display the file in new tab
+       * @param  {object} file
+       */
+      vm.viewFile = function (file) {
+        $rootScope.$broadcast('ct-spinner-show');
+        fileService.openFile(file).then(function (blobFile) {
+          $rootScope.$broadcast('ct-spinner-hide');
+        });
       };
 
+      (function init () {
```

Comment:
Init is way too far from the top. Please follow the guidelines https://github.com/johnpapa/angular-styleguide/tree/master/a1#controller-activation-promises